pkg/k8s/resources/resource.go

```diff
@@ -0,0 +1,16 @@
+package resources
+
+import "sigs.k8s.io/controller-runtime/pkg/client"
+
+// ResourceSpec allows generic Reconciler's implementation to get Hash value of the CR's Spec.
+// Hash is used to determine if spec has changed since the last status update or not.
+type ResourceSpec interface {
```

Comment:
I suggest just naming this Spec (& similarly for Status), this is more idiomatic in Golang and reads better when you type `resources.Spec` as opposed to `resources.ResourceSpec`.

Reply:
The package is 'controllers' so it is not obvious which spec is that. Also, I already have ResourceStatus - so it kinda consistent now.